content/en/docs/concepts/services-networking/endpoint-slices.md

```diff
@@ -58,36 +73,94 @@ endpoints:
+Endpoint Slices managed by the Endpoint Slice controller will contain the
+following information when it's available:
+
+* `kubernetes.io/hostname` - The name of the Node this endpoint is on.
+* `topology.kubernetes.io/zone` - The zone this endpoint is in.
+* `topology.kubernetes.io/region` - The region this endpoint is in.
+
+### Management
+
+By default, Endpoint Slices will be created and managed by the Endpoint Slice
+controller, but they can be useful beyond that. Endpoint Slices may be useful
+for a variety of additional use cases such as service mesh implementations. To
+ensure that multiple entities can manage Endpoint Slices without interfering
+with each other, a `endpointslice.kubernetes.io/managed-by` label is used to
+indicate the entity managing an Endpoint Slice. The Endpoint Slice controller
+sets `endpointslice-controller.k8s.io` as the value for this label on all
+Endpoint Slices it manages. Other entities managing Endpoint Slices should also
+set a unique value for this label.
+
+## Endpoint Slice Controller
+
+The Endpoint Slice controller watches Services and Pods to ensure corresponding
+Endpoint Slices are up to date. The controller will manage Endpoint Slices for 
+every Service with a Selector specified. These will represent the IPs of Pods
+matching the Service Selector. 
+
+### Size of Endpoint Slices
+
+By default, Endpoint Slices will be limited to a size of 100 endpoints each. This
+can be configured with the `--max-endpoints-per-slice` Controller Manager flag
+up to a maximum of 1000.
+
+### Distribution of Endpoint Slices
+
+Each Endpoint Slice has a set of ports that applies to all endpoints within the
+resource. When named ports are used for a Service, Pods may end up with
+different target port numbers for the same named port, requiring different
+Endpoint Slices. This is similar to the logic behind how subsets are grouped
+with Endpoints.
+
+The Controller tries to fill Endpoint Slices as full as possible, but does not
+actively rebalance them. The logic of the Controller is fairly straightforward:
+
+1. Iterate through existing Endpoint Slices, remove endpoints that are no longer
+   desired and update matching endpoints that have changed.
+2. Iterate through Endpoint Slices that have been modified in the first step and
+   fill them up with any new endpoints needed.
+3. If there's still new endpoints left to add, try to fit them into a previously
+   unchanged slice and/or create new ones.
+   
+Importantly, the third step prioritizes limiting Endpoint Slice updates over a
+perfectly full distribution of Endpoint Slices. As an example, if there are 10
+new endpoints to add and 2 Endpoint Slices with room for 5 more endpoints each,
+this approach will create a new Endpoint Slice instead of filling up the 2
+existing Endpoint Slices. In other words, a single Endpoint Slice creation is
+preferrable to multiple Endpoint Slice updates.
```

Comment:
A suggestion about this part, maybe we could take some typical examples for explaining the scales logical,  a good example is [pod qos](https://github.com/kubernetes/website/commit/f83de3f601bf853e92bf742236c3a331d226cadb#diff-43e0a44ed6b2437d8a5d7480fc2b58f0R100).

Reply:
I like that idea, good examples are always helpful when I'm trying to learn something new. I've been struggling with how to communicate that well with EndpointSlices where the scale is so significant. YAML snippets generally don't seem to work here because they'd get really long pretty quickly here, do you have any ideas for how I could write examples better here?

Reply:
Yes, no need to write examples by yaml, I think something like [this](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#example-states), use list of items to explain the cases and the behavior of EndpointSlice should be clear enough for our users.